amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/records/ExportRecordsTests.java

```diff
@@ -0,0 +1,221 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+import org.alfresco.rest.rm.community.base.BaseRMRestTest;
+import org.alfresco.rest.rm.community.model.record.Record;
+import org.alfresco.rest.rm.community.model.recordcategory.RecordCategory;
+import org.alfresco.rest.rm.community.model.recordcategory.RecordCategoryChild;
+import org.alfresco.rest.rm.community.requests.gscore.api.RecordFolderAPI;
+import org.alfresco.rest.v0.ExportAPI;
+import org.alfresco.rest.v0.RecordCategoriesAPI;
+import org.alfresco.rest.v0.RecordFoldersAPI;
+import org.alfresco.rest.v0.RecordsAPI;
+import org.alfresco.test.AlfrescoTest;
+import org.alfresco.utility.Utility;
+import org.alfresco.utility.report.log.Step;
+import org.apache.http.HttpResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * This class contains tests for
+ * Export REST API of RM Site
+ *
+ * @author Shubham Jain
+ * @since 3.5.0
+ */
+public class ExportRecordsTests extends BaseRMRestTest
+{
+    private RecordCategory rootCategory;
+
+    private RecordCategoryChild recordFolder;
+
+    private Record record;
+
+    private List<Record> recordList = new ArrayList<>();
+
+    @Autowired
+    private ExportAPI exportAPI;
+
+    @Autowired
+    private RecordCategoriesAPI categoriesAPI;
+
+    @Autowired
+    private RecordsAPI recordsAPI;
+
+    @BeforeClass (alwaysRun = true)
+    public void ExportRecordsTestsBeforeClass()
+    {
+        STEP("Create root level category");
+        rootCategory = createRootCategory(getRandomName("Category"));
+
+        STEP("Create the record folder inside the rootCategory");
+        recordFolder = createRecordFolder(rootCategory.getId(), getRandomName("Folder"));
+
+    }
+
+    @AfterTest
+    public void ExportRecordsTestsAfterTest(){
+
+        STEP("Delete the created Records");
+        for (int i = 0; i < recordList.size(); i++)
+        {
+            recordsAPI.deleteRecord(getAdminUser().getUsername(), getAdminUser().getPassword(), recordList.get(i).getName(),
+                    rootCategory.getName(), recordFolder.getName());
+
+        }
+        recordList.clear();
+    }
+
+    @AfterClass
+    public void ExportRecordsTestsAfter()
+    {
+
+        STEP("Delete the created folder");
+        deleteRecordFolder(recordFolder.getId());
+
+        STEP("Delete the created rootCategory");
+        categoriesAPI.deleteCategory(getAdminUser().getUsername(), getAdminUser().getPassword(), rootCategory.getName());
+
+    }
+
+
+    @DataProvider (name = "CreateRecordsInRecordFolder")
```

Comment:
Not needed

Reply:
As mentioned above, I have restructured the tests with a common Data Provider function and hence Data Provider will be needed.